vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_anon_card.cc

```diff
@@ -12,73 +12,17 @@
 #include "base/values.h"
 #include "bat/ledger/internal/contribution/contribution_anon_card.h"
 #include "bat/ledger/internal/ledger_impl.h"
-#include "bat/ledger/internal/request/request_sku.h"
-#include "bat/ledger/internal/request/request_util.h"
-#include "bat/ledger/internal/response/response_sku.h"
 #include "bat/ledger/internal/sku/sku_util.h"
 
 using std::placeholders::_1;
 using std::placeholders::_2;
 using std::placeholders::_3;
 
-namespace {
-
-std::string GetTransactionPayload(
-    const double amount,
-    const std::string& order_id,
-    const std::string& destination,
-    const std::string& payment_id,
-    const std::vector<uint8_t>& seed) {
-  base::Value denomination(base::Value::Type::DICTIONARY);
-  denomination.SetStringKey("amount", base::StringPrintf("%g", amount));
-  denomination.SetStringKey("currency", "BAT");
-
-  base::Value octets(base::Value::Type::DICTIONARY);
-  octets.SetKey("denomination", std::move(denomination));
-  octets.SetStringKey("destination", destination);
-
-  std::string octets_json;
-  base::JSONWriter::Write(octets, &octets_json);
-
-  const auto sign_headers = braveledger_request_util::GetSignHeaders(
-      order_id,
-      octets_json,
-      "primary",
-      seed,
-      true);
-
-  base::Value headers(base::Value::Type::DICTIONARY);
-  headers.SetStringKey("digest", sign_headers.at("digest"));
-  headers.SetStringKey("idempotency-key", order_id);
-  headers.SetStringKey("signature", sign_headers.at("signature"));
-
-  base::Value transaction(base::Value::Type::DICTIONARY);
-  transaction.SetKey("headers", std::move(headers));
-  transaction.SetStringKey("octets", octets_json);
-
-  std::string transaction_json;
-  base::JSONWriter::Write(transaction, &transaction_json);
-  std::string transaction_base64;
-  base::Base64Encode(transaction_json, &transaction_base64);
-
-  base::Value body(base::Value::Type::DICTIONARY);
-  body.SetStringKey("paymentId", payment_id);
-  body.SetStringKey("kind", braveledger_sku::ConvertTransactionTypeToString(
-      ledger::SKUTransactionType::ANONYMOUS_CARD));
-  body.SetStringKey("transaction", transaction_base64);
-
-  std::string body_json;
-  base::JSONWriter::Write(body, &body_json);
-
-  return body_json;
-}
-
-}  // namespace
-
 namespace braveledger_contribution {
 
 ContributionAnonCard::ContributionAnonCard(bat_ledger::LedgerImpl* ledger) :
-    ledger_(ledger) {
+    ledger_(ledger),
+    payment_server_(new ledger::endpoint::PaymentServer(ledger)) {
```

Comment:
Is there a reason to use `new` instead of `std::make_unique` here?

Reply:
not really, in first version of endpoint we used new and I then just copy pasted all around

Reply:
changing to `std::make_unique` in whole folder